src/main/java/core/basesyntax/Storage.java

```diff
@@ -0,0 +1,25 @@
+package core.basesyntax;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class Storage {
+
+    private Map<String, Integer> fruits = new HashMap<>();
+
+    public void setFruitQuantity(String fruit, int quantity) {
+        fruits.put(fruit, quantity);
+    }
+
+    public void addFruitQuantity(String fruit, int quantity) {
+        fruits.put(fruit, fruits.getOrDefault(fruit, 0) + quantity);
+    }
+
+    public void reduceFruitQuantity(String fruit, int quantity) {
+        fruits.put(fruit, fruits.getOrDefault(fruit, 0) - quantity);
+    }
```

Comment:
You should check if the quantity of the fruits is not negative after reducing it. If it is, you should throw an exception.